File: frontend/src/types/detection.ts
export type BoundingBox = {
  x1: number;
  y1: number;
  x2: number;
  y2: number;
};

export type Detection = {
  track_id?: number | null;
  label: string;
  confidence: number;
  bbox: BoundingBox;
};

export type DetectionResult = {
  camera_id: string;
  timestamp: string;
  detections: Detection[];
};

File: backend/app/schemas/detection.py
from datetime import datetime

from pydantic import BaseModel


class BoundingBox(BaseModel):
    x1: float
    y1: float
    x2: float
    y2: float


class Detection(BaseModel):
    track_id: int | None = None
    label: str
    confidence: float
    bbox: BoundingBox


class DetectionResult(BaseModel):
    camera_id: str
    timestamp: datetime
    detections: list[Detection]
